### PR TITLE
fix: retain objects which failed to reconcile

### PR DESCRIPTION
**Pull request by sdowell**
This change updates the set-inventory logic to retain objects which
failed to reconcile. This ensures that if you run the applier/destroyer
multiple times, an object that is failing to reconcile will be retained
in the inventory. Before this change, an object failing to reconcile
could be lost after multiple attempts (e.g. multiple destroys).

### CODE REVIEW

**Comment by sdowell** [comment]
This change is based on https://github.com/kubernetes-sigs/cli-utils/pull/622

File: pkg/apply/task/inv_set_task.go [hunk 1]
```diff
@@ -54,79 +83,113 @@ func (i *InvSetTask) Identifiers() object.ObjMetadataSet {
...
+	// then keep it in the inventory so it can be applied/pruned next time.
+	// It's likely that all the skipped deletes are already in the inventory,
+	// because the set of resources to prune comes from the inventory,
+	// but we're doing the intersection anyway just to be sure.
+	pruneSkips := i.PrevInventory.Intersection(im.SkippedDeletes())
+	klog.V(4).Infof("keep in inventory %d skipped prunes", len(pruneSkips))
+	invObjs = invObjs.Union(pruneSkips)
+
+	// If an object failed to reconcile and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
```
Comment (karlkfi) [inline pkg/apply/task/inv_set_task.go:135]: so it can be waited on next time

File: pkg/apply/task/inv_set_task_test.go [hunk 2]
```diff
@@ -160,6 +162,16 @@ func TestInvSetTask(t *testing.T) {
 			invalidObjs:   object.ObjMetadataSet{idInvalid},
 			expectedObjs:  object.ObjMetadataSet{id3},
 		},
+		"object failed to reconcile": {
```
Comment (karlkfi) [inline pkg/apply/task/inv_set_task_test.go:165]: can you add permutations for both apply and delete reconcile failures?

Comment (sdowell) [reply pkg/apply/task/inv_set_task_test.go]: done

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**